### PR TITLE
[risk=no][no ticket] Add and use Interactive package from Terra UI

### PR DESCRIPTION
**Pull request by dmohs**
This PR continues a series of changes to enable code sharing between the Terra and All of Us UIs. These changes include [isolating and packaging components](https://github.com/DataBiosphere/terra-ui/pull/3992) and [publishing them to an internal cache](https://github.com/DataBiosphere/terra-ui/pull/4103).

This change imports the shared `Interactive` component, replacing our own and eliminating the code.

Mostly authored by @nawatts.

Tested `yarn run build` and `yarn start` with some basic UI interactions.

Terra-UI exported components can be found here: https://github.com/DataBiosphere/terra-ui/tree/dev/packages/components

---

**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [x] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [x] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [x] None of the above apply to this change.

### CODE REVIEW

File: ui/config-overrides.js [hunk 1]
```diff
@@ -31,6 +31,25 @@ module.exports = {
...
+    /*
+      ERROR in ./node_modules/@terra-ui-packages/components/lib/es/index.js 1:0-44
+      Module not found: Error: Can't resolve './Interactive' in '/home/dmohs/b/r/aou/interactive/ui/node_modules/@terra-ui-packages/components/lib/es'
+      Did you mean 'Interactive.js'?
+      BREAKING CHANGE: The request './Interactive' failed to resolve only because it was resolved as fully specified
+      (probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
+      The extension in the request is mandatory for it to be fully specified.
+      Add the extension to the request.
+    */
+    config.module.rules[1].oneOf.forEach(rule => {
```
Comment (nawatts) [inline ui/config-overrides.js:44]: Updating to `@terra-ui-packages/components` v0.0.4 should allow removing this workaround.

**Comment by Qi77Qi** [comment]
fyi @jdcanas about this PR

**Review by Peter-Lavigne — APPROVED** [review]

File: ui/.npmrc [hunk 2]
```diff
@@ -0,0 +1,3 @@
+@fortawesome:registry=https://npm.fontawesome.com/
```
Comment (dmohs) [inline ui/.npmrc:1]: This line was one of two in gs://all-of-us-workbench-test-credentials/.npmrc that we copied to our environment. One of those lines was the authentication token needed for this package. I've moved that secret into its own file so the rest of this non-secret file could be version-controlled.

**Comment by yonghaoy** [comment]
can we also update README file here: https://github.com/all-of-us/workbench#ui-1?

**Comment by dmohs** [comment]
@yonghaoy Thank you! Great catch. I made https://precisionmedicineinitiative.atlassian.net/browse/RW-10815.

**Pull request merged**